modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java

```diff
@@ -250,70 +250,77 @@ public RustClientCodegen() {
+            processComposedSchemas(model, mdl, "oneOf");
+            processComposedSchemas(model, mdl, "anyOf");
+        }
 
-            Map<String, String> refsMapping = Optional.ofNullable(model.getDiscriminator())
-                    .map(Discriminator::getMapping).orElse(Collections.emptyMap());
-
-            // Reverse mapped references to use as baseName for oneOF, but different keys may point to the same $ref.
-            // Thus, we group them by the value
-            Map<String, List<String>> mappedNamesByRef = refsMapping.entrySet().stream()
-                    .collect(Collectors.groupingBy(Map.Entry::getValue,
-                            Collectors.mapping(Map.Entry::getKey, Collectors.toList())
-                    ));
-
-            for (int i = 0; i < newOneOfs.size(); i++) {
-                CodegenProperty oneOf = newOneOfs.get(i);
-                Schema schema = schemas.get(i);
-
-                if (mappedNamesByRef.containsKey(schema.get$ref())) {
-                    // prefer mapped names if present
-                    // remove mapping not in order not to reuse for the next occurrence of the ref
-                    List<String> names = mappedNamesByRef.get(schema.get$ref());
-                    String mappedName = names.remove(0);
-                    oneOf.setBaseName(mappedName);
-                    oneOf.setName(toModelName(mappedName));
-                } else if (!org.apache.commons.lang3.StringUtils.isEmpty(schema.get$ref())) {
-                    // use $ref if it's reference
-                    String refName = ModelUtils.getSimpleRef(schema.get$ref());
-                    if (refName != null) {
-                        String modelName = toModelName(refName);
-                        oneOf.setName(modelName);
-                        oneOf.setBaseName(refName);
-                    }
-                } else if (oneOf.isArray) {
-                    // If the type is an array, extend the name with the inner type to prevent name collisions
-                    // in case multiple arrays with different types are defined. If the user has manually specified
-                    // a name, use that name instead.
-                    String collectionWithTypeName = toModelName(schema.getType()) + oneOf.containerTypeMapped + oneOf.items.dataType;
-                    String oneOfName = Optional.ofNullable(schema.getTitle()).orElse(collectionWithTypeName);
-                    oneOf.setName(oneOfName);
-                }
-                else {
-                    // In-placed type (primitive), because there is no mapping or ref for it.
-                    // use camelized `title` if present, otherwise use `type`
-                    String oneOfName = Optional.ofNullable(schema.getTitle()).orElseGet(schema::getType);
-                    oneOf.setName(toModelName(oneOfName));
+        return mdl;
+    }
+
+    private void processComposedSchemas(Schema model, CodegenModel mdl, String type) {
```

Comment:
thanks for the PR

what about adding some comments (e.g. docstring, inline comments) for this function?